webapp/sources/rudder/rudder-core/src/main/resources/hooks.d/readme.adoc

```diff
@@ -30,7 +30,17 @@ you to use the patter "NN-hookname", with NN a number like
 - 32-63  : warning, warning log in /var/log/rudder/webapp/, continue to next hook
 - 64-255 : reserved for future use case. Behavior may change without notice.
 
+*Hooks have a maximum execution time allowed*. After some time (2s for policy generation node specific events, 30s for
+other cases) a warning will be issued. After a longer time (20s for policy generation node specific
+events, 5 min for other cases), the hook will be killed. You can change these timeouts with the following
+directives, that must be in the 1000 first characters of your hook script:
```

Comment:
You can change these timeouts with the following directives, that must be in the 1000 first characters of your hook script, prefixed by any type of comments characters (only the end of line is considered)